svm-decode/src/deserialize.rs

```diff
@@ -28,6 +40,121 @@ pub enum DynType {
     Option(Box<DynType>),
 }
 
+#[cfg(feature = "pyo3")]
+impl<'py> pyo3::FromPyObject<'py> for DynType {
+    fn extract_bound(ob: &pyo3::Bound<'py, pyo3::PyAny>) -> pyo3::PyResult<Self> {
+        use pyo3::types::PyAnyMethods;
+
+        let variant_str: String = ob.to_string();
+        let variant_str: &str = variant_str.as_str();
```

Comment:
I thought this was a bit hacky.
I couldn't find a method in the [pyo3::Bound](https://docs.rs/pyo3/latest/pyo3/struct.Bound.html) that I could later match, so I'm parsing it as a String.
If you look at the `svm_decode.__init__.py`, I had to implement a bunch of `__str__` methods so it doesn't include than just the name. For example:
```
    def __str__(self):
        return "FixedArray"
```
This may cause issues if the user tries to print the class.